corehq/apps/es/case_search.py

```diff
@@ -141,21 +141,32 @@ def case_property_filter(case_property_name, value):
     )
 
 
-def case_property_query(case_property_name, value, fuzzy=False):
+def case_property_query(case_property_name, value, fuzzy=False, mode=None):
     """
     Search for all cases where case property with name `case_property_name`` has text value `value`
     """
     if value is None:
         raise TypeError("You cannot pass 'None' as a case property value")
+    if mode not in ['AND', 'OR', None]:
+        raise ValueError(" 'mode' must be one of: 'AND', 'OR', None")
     if value == '':
         return case_property_missing(case_property_name)
-    if fuzzy:
+    if fuzzy and (mode is None or mode == 'OR'):
         return filters.OR(
             # fuzzy match
             case_property_text_query(case_property_name, value, fuzziness='AUTO'),
             # non-fuzzy match. added to improve the score of exact matches
             case_property_text_query(case_property_name, value),
         )
+    if fuzzy and mode == 'AND':
+        return filters.AND(
```

Comment:
This should still be an OR. We want: "does it match exactly OR does it fuzzy match"

Reply:
I see--fixed